### PR TITLE
feat(recommend): introduce `connectFrequentlyBoughtTogether` connector

### PR DESCRIPTION
**Pull request by raed667**
This PR introduces the `connectFrequentlyBoughtTogether` connector in InstantSearch.js.

FX-2794

### CODE REVIEW

**Review by dhayab — APPROVED** [review]
Following our last sync, I'm fine with the state of this PR, with the listed follow ups:

- separate PR to properly handle multi objectIDs within the state (#6138)
- separate PR to switch the connector's unit tests to the CTS (waiting for approval on the proposal in #6142, and pre-requisite to FX-2758)

**Review by sarahdayan — APPROVED** [review]

**Pull request merged**
